lesson3.md

```diff
@@ -0,0 +1,841 @@
+
+---
+
+
+### Operator chaining (cont.)
+
+Let's break it down:
+
+```js
+7 > 6 > 5
+```
+
+```js
+(7 > 6) > 5   // 7 > 6 is true
+```
+<!-- .element: class="fragment" -->
+
+```js
+true > 5      // meaningless comparison!
+```
+<!-- .element: class="fragment" -->
+
+Don't chain operators that return a boolean!
+
+
+---
+
+
+BUT what if we want to check if a number is smaller than 7 and bigger than 5?
+
+```js
+7 > myNumber > 5  // what do we do now???
+```
+
+
+Logical Operators! <!-- .element: class="fragment" -->
+
+
+---
+
+
+<!-- .slide: id="lesson3:operators" -->
+
+
+### Logical Operators
+
+These operators only make sense on `boolean` values:
+
+* `&&` Logical AND
+* `||` Logical OR
+* `!` Logical NOT
+
+
+---
+
+
+### Logical AND &&
+
+AND takes two booleans and returns `true` if **both** booleans are `true`
+
+```javascript
+true  && true   // true
+true  && false  // false
+false && true   // false
+false && false  // false
+```
+
+
+---
+
+
+### Example - AND &&
+
+I need bread AND cheese to make a cheese toast
+
+```js
+let hasCheese = true;
+let hasBread = true;
+
+let canMakeCheeseToast = hasCheese && hasBread;
+console.log(canMakeCheeseToast); // ?
+
+```
+
+true!
+<!-- .element: class="fragment" -->
+
+
+---
+
+
+### Logical OR ||
+
+OR takes two booleans and returns `true` if **at least one** is `true`
+
+```javascript
+true  || true   // true
+true  || false  // true
+false || true   // true
+false || false  // false
+```
+
+
+---
+
+
+
+### Example - OR ||
+
+To apply to a job, I need to know German OR English
+
+```js
+let knowsGerman = false;
+let knowsEnglish = false;
+
+let canApply = knowsGerman || knowsEnglish;
+console.log(canApply); // ?
+
+```
+
+false!
+<!-- .element: class="fragment" -->
+
+
+---
+
+
+### Let's practice
+
+Use JavaScript to compute:
+
+* Am I the youngest person in the group?
+* Am I younger than at least one person in the group?
+
+
+The group is you, the person to your left and the one to your right.
+
+
+---
+
+
+### Solution
+
+Lets assume my age is 42.
+
+Am I the youngest person in the group?
+
+I need to be younger than person to my left AND I need to be younger than person to my right
+
+```js
+(42 < 20) && (42 < 35)  // false - I am not the youngest person
+```
+
+
+---
+
+
+Am I younger than at least one person in the group?
+
+I need to be younger than person to my left OR I need to be younger than person to my right
+
+```js
+(42 < 20) || (42 < 35)
+// false - I am not younger than at least one other person
+```
+
+
+---
+
+
+### Logical NOT !
+
+Logical NOT negates a boolean:
+
+```javascript
+!true  // false
+!false // true
+```
+
+Example: Am I NOT the youngest person?
+
+
+---
+
+
+### Operator precedence
+
+What do you think is the result?
+
+```js
+2 + 2 * 2
+```
+
+Result: `6` <!-- .element: class="fragment" -->
+
+
+---
+
+
+* Operators have priorities
+* Operators with higher precedence are evaluated first.
+* For example: multiplication `*` is more important than addition `+`
```

Comment:
I think we should stick to one of these words.
Operators either have a priority, a precedence or an importance.

Reply:
I see your point, but I think it is still important to use simple synonyms for the word.

Precedence is the technical term and we assume it is new to the students, but priority and importance are very common.